File: Jump Game/jump game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxReachable = nums[0]
        i = 1
        while (i < len(nums) and maxReachable >= i):
            if (i + nums[i] > maxReachable):
                maxReachable = i + nums[i]
            i += 1
            
        if (maxReachable >= len(nums) - 1):
            return True
        else:
            return False

File: Jump Game/jump game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxReachable = nums[0];
        
        int i = 1;
        while (i < nums.size() && maxReachable >= i) {
            if (i + nums[i] > maxReachable) {
                maxReachable = i + nums[i];
            }
            i++;
        }
        
        if (maxReachable >= nums.size() - 1) {
            return true;
        }
        else {
            return false;
        }
    }
};